### PR TITLE
Update CMake config to generate working Visual Studio solutions

### PR DESCRIPTION
**Pull request by addictgamer**
Objectives:
 - [x] cmake can generate VS files that compile
 - [x] make INSTALL.txt slightly less terrible (now INSTALL.md)
 - [x] make the editor work too
 - [x] link README.md to INSTALL.md (I probably should redo README.md in general at some point...)
 - [x] cmake can generate FMOD VS files
 - [x] cmake can generate Steamworks VS files
 - [x] cmake can generate EOS VS files
 - [x] cmake can generate Steamworks + EOS VS files
 - [x] don't spawn the command prompt window
 - [x] verify no regressions to Mac or Linux builds
 - [x] get help from the community verifying the same for *BSD (~note to contributors: don't bother doing this until I've had a chance to test on and fix any issues in the Mac or Linux builds, because that would probably fix most BSD issues too.~ Ready!)
 - [ ] verification for other platforms would be great...don't know if any of them even work anymore?

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -1,8 +1,8 @@
-if( WIN32 )
-	list(APPEND GAME_SOURCES
-		"${CMAKE_CURRENT_SOURCE_DIR}/wingameicon.o"
-	)
-endif()
+# if( WIN32 )
+# 	list(APPEND GAME_SOURCES
+# 		"${CMAKE_CURRENT_SOURCE_DIR}/wingameicon.o"
+# 	)
+# endif()
```
Comment (addictgamer) [inline src/CMakeLists.txt:5]: Issues with this...need to fix and uncomment before merge

- [ ] fix game and editor icon

File: src/Config.hpp [hunk 2]
```diff
@@ -0,0 +1,60 @@
+
```
Comment (addictgamer) [inline src/Config.hpp:1]: oops.

- [ ] Uncommit Config.hpp!

**Comment by addictgamer** [comment]
Well that's a good sign, looks like CI was still able to build it...will need to check at my machine and verify it doesn't skip Steam or FMOD support :)

**Comment by addictgamer** [comment]
Looks like Linux and Mac are working fine after this change.

@devnexen could I ask you to take a quick look at BSD and make sure this doesn't break anything there?

**Comment by devnexen** [comment]
All good ;-)

**Pull request merged**